base/shell/explorer/traywnd.cpp

```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
 
         m_RunFileDlgOwner = hwnd;
 
-        RunFileDlg(hwnd, NULL, NULL, NULL, NULL, RFF_CALCDIRECTORY);
+        WCHAR szHomeDrive[MAX_PATH], szHomePath[MAX_PATH];
+        GetEnvironmentVariableW(L"HOMEDRIVE", szHomeDrive, _countof(szHomeDrive));
+        GetEnvironmentVariableW(L"HOMEPATH", szHomePath, _countof(szHomePath));
+
+        WCHAR szDefaultDir[MAX_PATH];
+        StringCchCopyW(szDefaultDir, _countof(szDefaultDir), szHomeDrive);
+        StringCchCatW(szDefaultDir, _countof(szDefaultDir), szHomePath);
```

Comment:
You can use CStringW variables instead:
- one that will contain the default directory "szDefaultDir", which is first initialized by a call to CStringW::GetEnvironmentVariable to initialize it with HOMEDRIVE;
- a second one that is initialized (again with CStringW::GetEnvironmentVariable) with HOMEPATH,
- then you concatenate both of these back into szDefaultDir (szDefaultDir += szHomePath),
- and you use (LPCWSTR)szDefaultDir as the 3rd parameter in RunFileDlg.

The resulting code should be cleaner and in the spirit of C++ used in shell/explorer.

Reply:
Thank you for reviewing. Done.